webclient/pas/views.py

```diff
@@ -600,12 +607,17 @@ def archivovat(request, ident_cely):
             status=403,
         )
     if request.method == "POST":
-        sn.create_transaction(request.user, SAMOSTATNY_NALEZ_ARCHIVOVAN)
-        sn.set_archivovany(request.user)
-        sn.igsn_publish()
-        sn.set_igsn()
-        sn.close_active_transaction_when_finished = True
-        sn.save()
+        fedora_transaction = sn.create_transaction(request.user, SAMOSTATNY_NALEZ_ARCHIVOVAN)
+        try:
+            sn.set_archivovany(request.user)
+            sn.igsn_publish()
+            sn.set_igsn()
+            sn.close_active_transaction_when_finished = True
+            sn.save()
```

Comment:
Tady může být myslím také problém, když to selže na něčem jiném, než na přidělení igsn, protože to zůstane publikované, ale záznam se neposune...